Releasing.md

```diff
@@ -213,29 +211,21 @@ For the body of the post, just copy this checklist and again replace all occurre
 <!-- /wp:paragraph -->
 
 <!-- wp:heading {"level":3} -->
-<h3>Sync the Release to Development Branches</h3>
+<h3>Merge Release Branches</h3>
 <!-- /wp:heading -->
 
-<!-- wp:paragraph -->
-<p>o If there are any conflicts in the gutenberg PR, merge <code>trunk</code> into it and resolve them.</p>
```

Comment:
I'm happy to add this back but it feels like a redundant step. I can't imagine a release wrangler not knowing how to resolve a merge to trunk conflict. At the the very least we can streamline the checklist by merging the info here into the "Merge the gutenberg PR ..." step